pkg/condition/types/ipaddress.go

```diff
@@ -0,0 +1,129 @@
+	openfgav1.ConditionParamTypeRef_TYPE_NAME_IPADDRESS,
+	cel.ObjectType("IPAddress"),
+	ipaddressTypeConverterFunc,
+	cel.Function("in_cidr",
+		cel.MemberOverload("ipaddr_in_cidr",
+			[]*cel.Type{cel.ObjectType("IPAddress"), cel.StringType},
+			cel.BoolType,
+			cel.BinaryBinding(ipaddressCELBinaryBinding),
+		),
+	),
+)
+
+// IPAddress represents a network IP address.
+type IPAddress struct {
+	addr netip.Addr
+}
+
+// ParseIPAddress attempts to parse the provided ip string. If the provided string does
+// not define a well-formed IP address, then an error is returned.
+func ParseIPAddress(ip string) (IPAddress, error) {
+	addr, err := netip.ParseAddr(ip)
+	if err != nil {
+		return IPAddress{}, err
+	}
+
+	return IPAddress{addr}, nil
+}
+
+// ipaddrCelType defines a CEL type for the IPAddress and registers it as a receiver type.
+var ipaddrCelType = cel.ObjectType("IPAddress", traits.ReceiverType)
+
+// ConvertToNative implements the CEL ref.Val.ConvertToNative.
+//
+// See https://pkg.go.dev/github.com/google/cel-go/common/types/ref#Val
+func (ip IPAddress) ConvertToNative(typeDesc reflect.Type) (any, error) {
+
+	if reflect.TypeOf(ip).AssignableTo(typeDesc) {
+		return ip, nil
+	}
+
+	switch typeDesc {
+	case reflect.TypeOf(""):
+		return ip.addr.String(), nil
+	default:
+		return nil, fmt.Errorf("failed to convert from type '%s' to native Go type 'IPAddress'", typeDesc)
+	}
+}
+
+// ConvertToType implements the CEL ref.Val.ConvertToType.
+//
+// See https://pkg.go.dev/github.com/google/cel-go/common/types/ref#Val
+func (ip IPAddress) ConvertToType(typeValue ref.Type) ref.Val {
+
+	switch typeValue {
+	case types.StringType:
+		return types.String(ip.addr.String())
+	case types.TypeType:
+		return ipaddrCelType
+	default:
+		return types.NewErr("failed to convert from CEL type '%s' to '%s'", ipaddrCelType, typeValue)
+	}
+}
+
+// Equal implements the CEL ref.Val.Equal.
+//
+// See https://pkg.go.dev/github.com/google/cel-go/common/types/ref#Val
+func (ip IPAddress) Equal(other ref.Val) ref.Val {
+
+	otherip, ok := other.(IPAddress)
+	if !ok {
+		return types.NoSuchOverloadErr()
+	}
+
+	return types.Bool(ip.addr.Compare(otherip.addr) == 0)
+}
+
+// Type implements the CEL ref.Val.Type.
+//
+// See https://pkg.go.dev/github.com/google/cel-go/common/types/ref#Val
+func (ip IPAddress) Type() ref.Type {
+	return ipaddrCelType
+}
+
+// Value implements ref.Val.Value.
+//
+// See https://pkg.go.dev/github.com/google/cel-go/common/types/ref#Val
+func (ip IPAddress) Value() any {
+	return ip
+}
+
+// ipaddressBinaryBinding implements a cel.BinaryBinding that is used as a receiver overload for
+// comparing an ipaddress value against a network CIDR defined as a string. If the ipaddress is
+// within the CIDR range this binding will return true, otherwise it will return false or an error.
+//
+// See https://pkg.go.dev/github.com/google/cel-go/cel#BinaryBinding
+func ipaddressCELBinaryBinding(lhs, rhs ref.Val) ref.Val {
```

Comment:
Mind adding an explicit test for this in `ipaddress_test.go`?

These links should help with it:
https://pkg.go.dev/github.com/google/cel-go@v0.18.0/common/types/ref#Val
https://pkg.go.dev/github.com/google/cel-go@v0.18.0/common/types#String (implements ref.Val for string) - use this to mimic the CIDR string